In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

# 1. Graduate Admission Rate

Graduate admission rate: Multiple Linear Regression

In [12]:
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error

In [58]:
#Import data
admission_rate=pd.read_csv('/Users/mirandidallas-fuge/Desktop/Admission_Predict.csv')

y=admission_rate['Chance of Admit ']
x=admission_rate.iloc[:,1:8]

In [13]:
# Fit the Model
reg = LinearRegression().fit(x, y)
# Print the Score
reg.score(x, y)
print("Multiple Linear Regression score:", reg.score(x, y))
print("Multiple Linear Regression MSE:", mean_squared_error(y, reg.predict(x)))
print("Multiple Linear Regression coef:", reg.coef_)

Multiple Linear Regression score: 0.8034713719824393
Multiple Linear Regression MSE: 0.003986893243246914
Multiple Linear Regression coef: [ 0.00173741  0.00291958  0.00571666 -0.00330517  0.02235313  0.11893945
  0.02452511]


Graduate admission rate: Lasso regression

In [16]:
from sklearn import linear_model
from sklearn.metrics import mean_squared_error
from sklearn.preprocessing import StandardScaler

In [17]:
# Scale data 
scaler = StandardScaler()
scaler.fit(x)
X = scaler.transform(x)

In [18]:
# Fit the Model and Print Score
lasso = linear_model.Lasso()
lasso.fit(X, y)
print("lasso score:", lasso.score(X, y))
print("lasso MSE:", mean_squared_error(y, lasso.predict(X)))
print("lasso coef:", lasso.coef_)

lasso score: 0.0
lasso MSE: 0.0202865775
lasso coef: [0. 0. 0. 0. 0. 0. 0.]


Graduate admission rate: Ridge regression

In [27]:
# Fit the Model and Print Score
ridge = linear_model.Ridge()
ridge.fit(X, y)

Ridge(alpha=1.0, copy_X=True, fit_intercept=True, max_iter=None,
   normalize=False, random_state=None, solver='auto', tol=0.001)

In [29]:
print("ridge score:",ridge.score(X, y))
print("ridge MSE:", mean_squared_error(y, ridge.predict(X)))
print("ridge coef:", ridge.coef_)

ridge score: 0.8034661029324273
ridge MSE: 0.003987000134238337
ridge coef: [ 0.02012161  0.01784985  0.00665018 -0.00311891  0.02005063  0.0701277
  0.01220766]


### Summary for graduate school data: 
- Multiple Linear Regression - MSE 0.003986893243246914
- Lasso Regression - MSE 0.0202865775
- Ridge Regression - MSE 0.003987000134238337

Thus since we want the smallest MSE of these, linear regression performs the best.

##### Data set attributes: 
- small dataset, low dimension, all numerical attributes

# 2. Crime incidence in community

In [206]:
#Import data
crime = pd.read_csv('/Users/mirandidallas-fuge/Desktop/CommViolPredUnnormalizedData.txt')

Crime: multiple linear regression 

In [ ]:
# Fit the Model
reg = LinearRegression().fit(x, y)
# Print the Score
reg.score(x, y)
print("Multiple Linear Regression score:", reg.score(x, y))
print("Multiple Linear Regression MSE:", mean_squared_error(y, reg.predict(x)))
print("Multiple Linear Regression coef:", reg.coef_)

Crime: Lasso regression

In [ ]:
# Scale data 
scaler = StandardScaler()
scaler.fit(x)
X = scaler.transform(x)

In [ ]:
# Fit the Model and Print Score
lasso = linear_model.Lasso()
lasso.fit(X, y)
print("lasso score:", lasso.score(X, y))
print("lasso MSE:", mean_squared_error(y, lasso.predict(X)))
print("lasso coef:", lasso.coef_)

Crime: Ridge regression

In [ ]:
# Fit the Model and Print Score
ridge = linear_model.Ridge()
ridge.fit(X, y)

In [ ]:
print("ridge score:",ridge.score(X, y))
print("ridge MSE:", mean_squared_error(y, ridge.predict(X)))
print("ridge coef:", ridge.coef_)

### Summary for crime data: 
- Multiple Linear Regression - 
- Lasso Regression - 
- Ridge Regression - 

Thus since we want the smallest MSE of these, _________  performs the best.

##### Data set attributes: 
-

# 3. Automobile

Automobile: multiple linear regression 

In [112]:
#process missing values
import numpy as np
from sklearn.preprocessing import Imputer
cars=pd.read_csv('/Users/mirandidallas-fuge/Desktop/imports-85.data')

In [120]:
x = cars.iloc[:, 0:-1]
y = cars.iloc[:,-1]

fix missing values 

In [ ]:
# Fit the Model
reg = LinearRegression().fit(x, y)
# Print the Score
reg.score(x, y)
print("Multiple Linear Regression score:", reg.score(x, y))
print("Multiple Linear Regression MSE:", mean_squared_error(y, reg.predict(x)))
print("Multiple Linear Regression coef:", reg.coef_)

Automobile: Lasso regression

In [ ]:
# Scale data 
scaler = StandardScaler()
scaler.fit(x)
X = scaler.transform(x)

In [ ]:
# Fit the Model and Print Score
lasso = linear_model.Lasso()
lasso.fit(X, y)
print("lasso score:", lasso.score(X, y))
print("lasso MSE:", mean_squared_error(y, lasso.predict(X)))
print("lasso coef:", lasso.coef_)

Automobile: Ridge regression

In [ ]:
# Fit the Model and Print Score
ridge = linear_model.Ridge()
ridge.fit(X, y)

In [ ]:
print("ridge score:",ridge.score(X, y))
print("ridge MSE:", mean_squared_error(y, ridge.predict(X)))
print("ridge coef:", ridge.coef_)

### Summary for automobile data: 
- Multiple Linear Regression - 
- Lasso Regression - 
- Ridge Regression - 

Thus since we want the smallest MSE of these, _________  performs the best.

##### Data set attributes: 
-

# 4. Breast Cancer 

Breast Cancer: Multiple Linear Regression 

In [75]:
Breast_cancer_wisconsin = pd.read_csv('/Users/mirandidallas-fuge/Desktop/breast-cancer-wisconsin.data')

In [79]:
x=Breast_cancer_wisconsin.iloc[:,2:]

In [80]:
y=Breast_cancer_wisconsin.iloc[:,1]

fix missing values

In [98]:
from sklearn.preprocessing import Imputer
imputer = Imputer(missing_values='NaN', strategy='mean', axis=0)

imputer = imputer.fit(x[:,1:5])
x[:,1:5] = imputer.transform(x[:, 1:5])

imputer = imputer.fit(x)
x = imputer.transform(x)

In [ ]:
# Fit the Model
reg = LinearRegression().fit(x, y)
# Print the Score
reg.score(x, y)
print("Multiple Linear Regression score:", reg.score(x, y))
print("Multiple Linear Regression MSE:", mean_squared_error(y, reg.predict(x)))
print("Multiple Linear Regression coef:", reg.coef_)

Breast Cancer: Lasso

In [ ]:
# Scale data 
scaler = StandardScaler()
scaler.fit(x)
X = scaler.transform(x)

In [ ]:
# Fit the Model and Print Score
lasso = linear_model.Lasso()
lasso.fit(X, y)
print("lasso score:", lasso.score(X, y))
print("lasso MSE:", mean_squared_error(y, lasso.predict(X)))
print("lasso coef:", lasso.coef_)

Breast Cancer: Ridge Regression

In [ ]:
# Fit the Model and Print Score
ridge = linear_model.Ridge()
ridge.fit(X, y)

In [ ]:
print("ridge score:",ridge.score(X, y))
print("ridge MSE:", mean_squared_error(y, ridge.predict(X)))
print("ridge coef:", ridge.coef_)

### Summary for breast cancer data: 
- Multiple Linear Regression - 
- Lasso Regression - 
- Ridge Regression - 

Thus since we want the smallest MSE of these, _________  performs the best.

##### Data set attributes: 
-

# 5. Parkinsons Data 

In [7]:
Parkinsons_data = pd.read_csv('/Users/mirandidallas-fuge/Desktop/parkinsons_updrs.data')

In [134]:
y = Parkinsons_data['total_UPDRS']

In [135]:
x = Parkinsons_data.drop(['total_UPDRS'], axis=1)

Parkinsons: Multiple Linear Regression 

In [136]:
# Fit the Model
reg = LinearRegression().fit(x, y)
# Print the Score
reg.score(x, y)
print("Multiple Linear Regression score:", reg.score(x, y))
print("Multiple Linear Regression MSE:", mean_squared_error(y, reg.predict(x)))
print("Multiple Linear Regression coef:", reg.coef_)

Multiple Linear Regression score: 0.9100144198179335
Multiple Linear Regression MSE: 10.30124083758564
Multiple Linear Regression coef: [ 4.72741397e-02  7.43271063e-02 -1.75884711e+00  2.64879855e-03
  1.20660278e+00 -3.56842204e+02  1.77854322e+04  8.27931046e+03
  8.22633190e+01 -2.60542785e+03 -5.08475929e+01  6.92595138e-01
 -1.36712611e+04  1.06861484e+02 -4.25065839e+01  4.55455275e+03
 -4.26252046e+00 -8.81464413e-02  2.85995358e+00 -3.94378918e+00
 -3.95847026e+00]


Parkinsons: Lasso Regression 

In [137]:
# Scale data 
scaler = StandardScaler()
scaler.fit(x)
X = scaler.transform(x)

In [138]:
# Fit the Model and Print Score
lasso = linear_model.Lasso()
lasso.fit(X, y)
print("lasso score:", lasso.score(X, y))
print("lasso MSE:", mean_squared_error(y, lasso.predict(X)))
print("lasso coef:", lasso.coef_)

lasso score: 0.8885117507280906
lasso MSE: 12.762792705087232
lasso coef: [ 0.          0.         -0.          0.          9.13478072  0.
  0.          0.         -0.          0.          0.          0.
  0.          0.          0.          0.         -0.         -0.
  0.         -0.          0.        ]


Parkinsons: Ridge Regression 

In [139]:
# Fit the Model and Print Score
ridge = linear_model.Ridge()
ridge.fit(X, y)

Ridge(alpha=1.0, copy_X=True, fit_intercept=True, max_iter=None,
   normalize=False, random_state=None, solver='auto', tol=0.001)

In [140]:
print("ridge score:",ridge.score(X, y))
print("ridge MSE:", mean_squared_error(y, ridge.predict(X)))
print("ridge coef:", ridge.coef_)

ridge score: 0.909998204912202
ridge MSE: 10.303097064425131
ridge coef: [ 0.58416848  0.6559799  -0.81844061  0.14182767  9.80558099 -1.96009871
  0.63471423  0.76440024  0.29214967  0.65528174 -1.27777731  0.1376577
 -0.07298332  1.76981864 -0.85086344 -0.02896885 -0.25614925 -0.38072316
  0.28682972 -0.28015241 -0.36377144]


### Summary for parkinson data: 
- Multiple Linear Regression - 10.30124083758564
- Lasso Regression - 12.762792705087232
- Ridge Regression - 10.303097064425131

Thus since we want the smallest MSE of these, linear performs the best but due to the large MSE values non of them perform that well.

##### Data set attributes: 
- small data set

# 6. Mercedes-Benz Greener Manufacturing

In [157]:
Mercedes = pd.read_csv('/Users/mirandidallas-fuge/Desktop/train_mercedes.csv')
y = Mercedes['y']
x = Mercedes.drop(['y'], axis=1)

fix categorical data

Mercedes-Benz Greener Manufacturing: Multiple Linear Regression

In [ ]:
# Fit the Model
reg = LinearRegression().fit(x, y)
# Print the Score
reg.score(x, y)
print("Multiple Linear Regression score:", reg.score(x, y))
print("Multiple Linear Regression MSE:", mean_squared_error(y, reg.predict(x)))
print("Multiple Linear Regression coef:", reg.coef_)

Mercedes-Benz Greener Manufacturing: Lasso Regression 

In [ ]:
# Scale data 
scaler = StandardScaler()
scaler.fit(x)
X = scaler.transform(x)

In [ ]:
# Fit the Model and Print Score
lasso = linear_model.Lasso()
lasso.fit(X, y)
print("lasso score:", lasso.score(X, y))
print("lasso MSE:", mean_squared_error(y, lasso.predict(X)))
print("lasso coef:", lasso.coef_)

Mercedes-Benz Greener Manufacturing: Ridge Regression 

In [ ]:
# Fit the Model and Print Score
ridge = linear_model.Ridge()
ridge.fit(X, y)

In [ ]:
print("ridge score:",ridge.score(X, y))
print("ridge MSE:", mean_squared_error(y, ridge.predict(X)))
print("ridge coef:", ridge.coef_)

### Summary for Mercedes data: 
- Multiple Linear Regression - 
- Lasso Regression - 
- Ridge Regression - 

Thus since we want the smallest MSE of these, _____ performs the best

##### Data set attributes: 
- 

# 7. Russian Housing Prices 

In [165]:
russian = pd.read_csv('/Users/mirandidallas-fuge/Desktop/train_russian.csv')

In [166]:
y = russian['price_doc']
x = russian.drop(['price_doc'], axis=1)

fix missing values 

Russian: Multiple Linear Regression 

In [ ]:
# Fit the Model
reg = LinearRegression().fit(x, y)
# Print the Score
reg.score(x, y)
print("Multiple Linear Regression score:", reg.score(x, y))
print("Multiple Linear Regression MSE:", mean_squared_error(y, reg.predict(x)))
print("Multiple Linear Regression coef:", reg.coef_)

Russian: Lasso Regression

In [ ]:
# Scale data 
scaler = StandardScaler()
scaler.fit(x)
X = scaler.transform(x)

In [ ]:
# Fit the Model and Print Score
lasso = linear_model.Lasso()
lasso.fit(X, y)
print("lasso score:", lasso.score(X, y))
print("lasso MSE:", mean_squared_error(y, lasso.predict(X)))
print("lasso coef:", lasso.coef_)

Russian: Ridge Regression

In [ ]:
# Fit the Model and Print Score
ridge = linear_model.Ridge()
ridge.fit(X, y)

In [ ]:
print("ridge score:",ridge.score(X, y))
print("ridge MSE:", mean_squared_error(y, ridge.predict(X)))
print("ridge coef:", ridge.coef_)

### Summary for Russian data: 
- Multiple Linear Regression - 
- Lasso Regression - 
- Ridge Regression - 

Thus since we want the smallest MSE of these, ________ performs the best 

##### Data set attributes: 
- 

# 8. Solar Flares

In [185]:
solar = pd.read_csv('/Users/mirandidallas-fuge/Desktop/flare.data2', sep = " ")

In [199]:
x = solar.iloc[:,1:6]
y = solar.iloc[:,7]

Solar: Multiple Linear Regression

In [201]:
# Fit the Model
reg = LinearRegression().fit(x, y)
# Print the Score
reg.score(x, y)
print("Multiple Linear Regression score:", reg.score(x, y))
print("Multiple Linear Regression MSE:", mean_squared_error(y, reg.predict(x)))
print("Multiple Linear Regression coef:", reg.coef_)

Multiple Linear Regression score: 0.07079476730209577
Multiple Linear Regression MSE: 0.0851232049849271
Multiple Linear Regression coef: [0.06338076 0.02650041 0.02537617 0.42973347 0.        ]


Solar: Lasso Regression

In [202]:
# Scale data 
scaler = StandardScaler()
scaler.fit(x)
X = scaler.transform(x)

In [203]:
# Fit the Model and Print Score
lasso = linear_model.Lasso()
lasso.fit(X, y)
print("lasso score:", lasso.score(X, y))
print("lasso MSE:", mean_squared_error(y, lasso.predict(X)))
print("lasso coef:", lasso.coef_)

lasso score: 0.0
lasso MSE: 0.0916086156098969
lasso coef: [0. 0. 0. 0. 0.]


Solar: Ridge Regression 

In [204]:
# Fit the Model and Print Score
ridge = linear_model.Ridge()
ridge.fit(X, y)

Ridge(alpha=1.0, copy_X=True, fit_intercept=True, max_iter=None,
   normalize=False, random_state=None, solver='auto', tol=0.001)

In [205]:
print("ridge score:",ridge.score(X, y))
print("ridge MSE:", mean_squared_error(y, ridge.predict(X)))
print("ridge coef:", ridge.coef_)

ridge score: 0.07079472336923387
ridge MSE: 0.08512320900955576
ridge coef: [0.02029078 0.013006   0.00838167 0.06745725 0.        ]


### Summary for Solar data: 
- Multiple Linear Regression - 0.0851232049849271
- Lasso Regression - 0.0916086156098969
- Ridge Regression - 0.08512320900955576

Thus since we want the smallest MSE of these, multiple linear regression performs the best 

##### Data set attributes: 
- small data set